### PR TITLE
replace all occurrences of "node", nodeName to "node", klog.KRef("", nodeName)

### PR DESCRIPTION
**Pull request by yangjunmyfm192085**
Signed-off-by: JunYang <yang.jun22@zte.com.cn>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
replace all occurrences of "node", nodeName to "node", klog.KRef("", nodeName)

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
part of kubernetes/enhancements#1602
As it is related to log, change the log here to structured log.
#### Does this PR introduce a user-facing change?

```release-note
 "NONE"
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
 "NONE"
```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 1]
```diff
@@ -570,7 +569,7 @@ func pickOneNodeForPreemption(nodesToVictims map[string]*extenderv1.Victims) str
 		// Get earliest start time of all pods on the current node.
 		earliestStartTimeOnNode := util.GetEarliestPodStartTime(nodesToVictims[node])
 		if earliestStartTimeOnNode == nil {
-			klog.ErrorS(errors.New("earliestStartTime is nil for node"), "should not reach here", "node", node)
+			klog.ErrorS(nil, "EarliestStartTime is nil for node. Should not reach here.", "node", node)
```
Comment (chendave) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:572]: Just wondering what's these kind of change benefit us?

Comment (yangjunmyfm192085) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go]: Yeah. I understand and I fixed them all, thank you very much.

File: pkg/scheduler/framework/plugins/examples/stateful/stateful.go [hunk 2]
```diff
@@ -84,7 +84,7 @@ func (mp *MultipointExample) PreBind(ctx context.Context, state *framework.Cycle
 // New initializes a new plugin and returns it.
 func New(config *runtime.Unknown, _ framework.Handle) (framework.Plugin, error) {
 	if config == nil {
-		klog.Error("MultipointExample configuration cannot be empty")
+		klog.ErrorS(nil, "MultipointExample configuration cannot be empty")
```
Comment (draveness) [inline pkg/scheduler/framework/plugins/examples/stateful/stateful.go:87]: why do we need to change the function here

Comment (draveness) [reply pkg/scheduler/framework/plugins/examples/stateful/stateful.go:87]: and add nil as param

Comment (yangjunmyfm192085) [reply pkg/scheduler/framework/plugins/examples/stateful/stateful.go:87]: According to the following rules, Error will be replaced by ErrorS. the first param format  of ErrorS is "type error interface", we can use nil as the first param. the second param is message.
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging

Comment (chendave) [reply pkg/scheduler/framework/plugins/examples/stateful/stateful.go:87]: I think this is abused, 

> We are not replacing currently used logging library (klog) or the way in which it is used
> We are not preserving full compatibility of logging output
> We are not proposing structuring all logs in Kubernetes

I suggest to change only where is necessary and do bring us some benefit.

Comment (yangjunmyfm192085) [reply pkg/scheduler/framework/plugins/examples/stateful/stateful.go]: Sorry, I missed this message.
I respect your suggestion, although my point  is different.
I have fixed it. Thank you very much.
@chendave, any questions, feel free to contact me.

File: pkg/scheduler/framework/plugins/defaultbinder/default_binder.go [hunk 3]
```diff
@@ -48,7 +48,7 @@ func (b DefaultBinder) Name() string {
 
 // Bind binds pods to nodes using the k8s client.
 func (b DefaultBinder) Bind(ctx context.Context, state *framework.CycleState, p *v1.Pod, nodeName string) *framework.Status {
-	klog.V(3).InfoS("Attempting to bind pod to node", "pod", klog.KObj(p), "node", nodeName)
+	klog.V(3).InfoS("Attempting to bind pod to node", "pod", klog.KObj(p), "nodeName", nodeName)
```
Comment (serathius) [inline pkg/scheduler/framework/plugins/defaultbinder/default_binder.go:51]:
```suggestion
	klog.V(3).InfoS("Attempting to bind pod to node", "pod", klog.KObj(p), "node", klog.KRef("", nodeName))
```

**Comment by Huang-Wei** [comment]
@yangjunmyfm192085 are you still working on it. If so, could you also update other cases:

```
pkg/scheduler/framework/plugins/defaultbinder/default_binder.go
52:	klog.V(3).InfoS("Attempting to bind pod to node", "pod", klog.KObj(p), "node", nodeName)

pkg/controller/util/node/controller_utils.go
122:	klog.V(2).InfoS("Update ready status of pods on node", "node", nodeName)
172:	klog.V(2).InfoS("Recording event message for node", "event", event, "node", nodeName)

pkg/controller/nodelifecycle/scheduler/taint_manager.go
353:		klog.V(2).InfoS("Not all taints are tolerated after update for pod on node", "pod", podNamespacedName.String(), "node", nodeName)
```

**Comment by dims** [comment]
/assign @ahg-g 

NOTE: just a one line change so should be a quick review!

**Comment by Huang-Wei** [comment]
@dims we're waiting for the author's reply on https://github.com/kubernetes/kubernetes/pull/99292#issuecomment-966497576 - it's preferred to make similar changes in this PR.

Ping @yangjunmyfm192085.

**Comment by yangjunmyfm192085** [comment]
@Huang-Wei, Thanks for reminding, I think the suggestions have been modified by other contributors.

**Comment by Huang-Wei** [comment]
@yangjunmyfm192085 If I recall clearly, when I firstly review this PR, it's already off the topic of "optimizing usage of klog.ErrorS", right?

So my comment back then was to replace all occurrences of `"node", nodeName` to `"node", klog.KRef("", nodeName)`, which are still valid in latest codebase:

```
wei.huang1@wei-imac:~/gospace/src/k8s.io/kubernetes|master ⇒  ag "klog.*\"node\", nodeName"
pkg/scheduler/framework/plugins/defaultbinder/default_binder.go
52:	klog.V(3).InfoS("Attempting to bind pod to node", "pod", klog.KObj(p), "node", nodeName)

pkg/controller/util/node/controller_utils.go
120:	klog.V(2).InfoS("Update ready status of pods on node", "node", nodeName)
168:	klog.V(2).InfoS("Recording event message for node", "event", event, "node", nodeName)

pkg/controller/podgc/gc_controller.go
203:				klog.ErrorS(err, "Error while getting node", "node", nodeName)

pkg/controller/nodelifecycle/scheduler/taint_manager.go
353:		klog.V(2).InfoS("Not all taints are tolerated after update for pod on node", "pod", podNamespacedName.String(), "node", nodeName)
```

**Comment by yangjunmyfm192085** [comment]
Sorry, I misunderstood. Yeah, I get it now, and have updated the pr.
Could you please review again? thanks @Huang-Wei

**Comment by ahg-g** [comment]
can you please update the PR title

**Comment by yangjunmyfm192085** [comment]
@ahg-g I have updated, could you please review again, thanks

**Comment by Huang-Wei** [comment]
Waiting for @kubernetes/api-approvers  to /approve.

**Review by yangjunmyfm192085 — COMMENTED** [review]
/assign @cheftako
Could you help approve this pr? thanks

**Review by yangjunmyfm192085 — COMMENTED** [review]
/ping @cheftako

**Review by yangjunmyfm192085 — COMMENTED** [review]
/assign @deads2k

**Pull request merged**